app/endpoints/raffle.py

```diff
@@ -0,0 +1,738 @@
+            firstname=user.firstname, name=user.name, nickname=user.nickname
+        )
+        hyperion_raffle_logger.info(
+            f"Add_ticket_to_user: A pack of {type_ticket.pack_size} tickets of type {type_id} has been buyed by user {display_name}({user.id}) for an amount of {type_ticket.price}€. ({request_id})"
+        )
+
+        return tickets
+
+    except ValueError as error:
+        raise HTTPException(status_code=422, detail=str(error))
+
+    finally:
+        locker_set(redis_client=redis_client, key=redis_key, lock=False)
+
+
+# @router.delete(
+#     "/tombola/tickets/{ticket_id}",
+#     status_code=204,
+#     tags=[Tags.raffle],
+# )
+# async def delete_ticket(
+#     ticket_id: str,
+#     db: AsyncSession = Depends(get_db),
+#     user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.soli)),
+# ):
+#     """
+#     Delete a ticket.
+
+#     **The user must be a member of the group soli to use this endpoint**
+#     """
+
+#     ticket = await cruds_raffle.get_ticket_by_id(ticket_id=ticket_id, db=db)
+#     if not ticket:
+#         raise HTTPException(status_code=404, detail="Ticket not found")
+
+#     await cruds_raffle.delete_ticket(db=db, ticket_id=ticket_id)
+
+
+@router.get(
+    "/tombola/users/{user_id}/tickets",
+    response_model=list[schemas_raffle.TicketComplete],
+    status_code=200,
+    tags=[Tags.raffle],
+)
+async def get_tickets_by_userid(
+    user_id: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Get tickets of a specific user.
+
+    **Only admin users can get tickets of another user**
+    """
+    user_db = await cruds_users.get_user_by_id(db=db, user_id=user_id)
+    if user_db is None:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    if user_id == user.id or is_user_member_of_an_allowed_group(user, [GroupType.soli]):
+        tickets = await cruds_raffle.get_ticket_by_userid(user_id=user_id, db=db)
+        return tickets
+
+    else:
+        raise HTTPException(
+            status_code=403,
+            detail="Users that are not member of the group admin can only access the endpoint for their own user_id.",
+        )
+
+
+@router.get(
+    "/tombola/raffle/{raffle_id}/tickets",
+    response_model=list[schemas_raffle.TicketComplete],
+    status_code=200,
+    tags=[Tags.raffle],
+)
+async def get_tickets_by_raffleid(
+    raffle_id: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.soli)),
+):
+    """
+    Get tickets from a specific raffle.
+
+    **The user must be a member of the group soli to use this endpoint
+    """
+
+    tickets = await cruds_raffle.get_ticket_by_raffleid(raffle_id=raffle_id, db=db)
+
+    return tickets
+
+
+@router.get(
+    "/tombola/lots",
+    response_model=list[schemas_raffle.LotSimple],
+    status_code=200,
+    tags=[Tags.raffle],
+)
+async def get_lots(
```

Comment:
Since there is a crud that get prizes using a raffle_id, is there a usage for this endpoint?

Reply:
This endpoint is used by the frontend for now